invoices/invoiceregistry.go

```diff
@@ -789,13 +796,13 @@ func (i *InvoiceRegistry) NotifyExitHopHtlc(rHash lntypes.Hash,
 
 			i.notifyHodlSubscribers(
 				*NewSettleResolution(
-					invoice.Terms.PaymentPreimage, key, acceptHeight,
+					invoice.Terms.PaymentPreimage, key, acceptHeight, result,
```

Comment:
Small question mark here whether all the previously added htlcs should be settled with this same result? Perhaps add a comment detailing this.

Reply:
If this is the first time we are settling the htlc, the other htlcs will correctly be notified with `ResultSettled` (because they are also being settled for the first time). On replays of the htlc settle, the already settled htlcs will not notify, although there could be some race between the settle and replay?

I think the only way this can be inconsistent is if we settle a htlc and have a failure before we notify the other htlcs and then reply it. They will be notified with `ReplaySettle` even though they are being settled for the first time. 

I'll add a comment for this, it shouldn't be too much of an issue since I can't see us using settle result vs replaySettle for anything actionable.

Reply:
Ok, sounds good.